examples/step-10/step-10.cc

```diff
@@ -399,10 +405,12 @@ int main()
     {
       std::cout.precision(16);
 
-      Step10::gnuplot_output<2>();
+      unsigned int dim = 2;
```

Comment:
This must be a compile time constant. In other words,
```suggestion
      constexpr unsigned int dim = 2;
```
would work.